refinery_core/src/util.rs

```diff
@@ -37,30 +37,41 @@ pub fn find_migration_files(
         )
     })?;
 
-    let file_paths = WalkDir::new(location)
+    let file_paths: Vec<PathBuf> = WalkDir::new(location)
         .into_iter()
         .filter_map(Result::ok)
         .map(DirEntry::into_path)
-        // filter by migration file regex
-        .filter(
-            move |entry| match entry.file_name().and_then(OsStr::to_str) {
-                Some(file_name) => {
-                    if re.is_match(file_name) {
-                        true
+        .collect();
+
+    let mut occurence: HashSet<String> = HashSet::with_capacity(file_paths.len());
+    let mut result: Vec<PathBuf> = Vec::with_capacity(file_paths.len());
+
+    for entry in &file_paths {
```

Comment:
this way we are performing two iteration cycles, whereas that's unnecessary right?
We can do it all in one cycle, I think it's preferable accounting the amortization cost.